spec/components/candidate_interface/references_review_component_spec.rb

```diff
@@ -5,8 +5,8 @@
     reference = create(:reference, :not_requested_yet)
     result = render_inline(described_class.new(references: [reference]))
 
-    name_row = result.css('.govuk-summary-list__row')[1].text
-    email_row = result.css('.govuk-summary-list__row')[2].text
+    name_row = result.css('.govuk-summary-list__row')[2].text
+    email_row = result.css('.govuk-summary-list__row')[3].text
```

Comment:
One for the future, but it feels a bit brittle to me that these tests hardcode which row they look at? (And took me a few mins to realise what the mistake was, as the error was confusing).

I wonder if it'd make sense to have some kind of rspec helper for summary lists which asserts that a given key and value are next to each other, but doesn’t care which order they are in? 🤔

Reply:
YES

Reply:
Yeah I think we've got into a bad habit of relying on hard-coded orders like this. Maybe we need a helper that does something like this to extract the attribute-value pairs to assert against:

```
attrs = result.css('.govuk-summary-list__row dt').map(&:text)
values = result.css('.govuk-summary-list__row dd:first').map(&:text)
attrs.zip(values).to_h
```

We could get really fancy and wrap it up in a custom matcher.